web/src/sub_store.ts

```diff
@@ -40,6 +50,21 @@ export type StreamSubscription = (Stream & StreamSpecificNotificationSettings) &
     subscribers?: number[];
 };
 
+// These properties are added in `stream_data` when hydrating the streams and are not present in the data we get from the server.
+export type ExtraStreamAttrs = {
+    render_subscribers: boolean;
+    newly_subscribed: boolean;
+    subscribed: boolean;
+    previously_subscribed?: boolean;
+};
+
+// This is the actual type of subscription objects we use in the app.
+export type StreamSubscription = PartialBy<
+    ApiStreamSubscription,
+    "pin_to_top" | "email_address" | "in_home_view"
+> &
+    ExtraStreamAttrs;
```

Comment:
If we’re creating subscription objects without `pin_to_top`, `email_address`, or `in_home_view`, that seems like a bug?

Reply:
So we are for some reason also creating `sub` objects from `NeverSubscribedStream` type and never subscribed streams does not have these fields so some `sub` objects in theory will indeed not have these properties.

Related Discussion: [CZO](https://chat.zulip.org/#narrow/stream/6-frontend/topic/stream_data.20typescript.20migration/near/1615755).

Reply:
Hmm. If it's a stream you've never been subscribed to, we know `pin_to_top=false`, and `email_address` is I think `null` -- you don't have access to it? 

For `in_home_view`, I don't think we ever want to access it -- it might be better to figure out a way to drop it from objects when first accepting them from the server in favor of the `is_muted` field that we actually use.

Reply:
Yeah, we can probably delete `in_home_view` the same way we delete `subscribers` ?
https://github.com/zulip/zulip/blob/2db715030d86819de8493afb879fb5c9c5e43139/web/src/stream_data.js#L740

Reply:
Yeah, that seems like a good plan.

Reply:
Alright dropped `in_home_view` from `ApiStreamSubscription` interface also when we will have actual zod schemas for these types, the zod will automatically remove these fields for us.